python/benchmarks/array_ops.py

```diff
@@ -0,0 +1,34 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import numpy as np
+import pyarrow as pa
+
+
+class ScalarAccess(object):
+    n = 10 ** 5
+
+    def setUp(self):
+        self._array = pa.array(list(range(self.n)), type=pa.int64())
+
+    def time_getitem(self):
+        for i in range(self.n):
+            self._array[i]
+
+    def time_getitem_as_py(self):
+        for i in range(self.n):
+            self._array[i].as_py()
```

Comment:
I wonder if this should loop over the array with elements already indexed so that we don't have to compute the difference between `time_getitem` and `time_getitem_as_py` to see the overhead of `as_py`. Something like

```python
for el in self._array2:
    el.as_py()
```

Reply:
Then we'll be benchmarking array iteration ;-) Instead I think I'll store the list of indexed values in the setup() method.

Reply:
Fair enough.

Reply:
Note we may want to benchmark `as_py()` on different types.
(though the inner machinery may already be exercised as part of `to_pylist`?)